server/client.go

```diff
@@ -770,6 +771,20 @@ func (c *client) processSub(argo []byte) (err error) {
 		return fmt.Errorf("processSub Parse Error: '%s'", arg)
 	}
 
+	// If dead-letter queues are enabled, check if it's a DLQ subscription,
+	// indicated by a subject starting with _SYS.dlq.
+	if atomic.LoadUint32(&c.srv.dlq) == 1 &&
+		sub.subject[0] == '_' && len(sub.subject) > 8 &&
+		sub.subject[1] == 'S' && sub.subject[2] == 'Y' &&
+		sub.subject[3] == 'S' && sub.subject[4] == '.' &&
+		sub.subject[5] == 'd' && sub.subject[6] == 'l' &&
+		sub.subject[7] == 'q' && sub.subject[8] == '.' {
+
```

Comment:
I was surprised by the way you implement the 'hasprefix' part, so I did some benchmarks (see https://gist.github.com/cdevienne/5c9ce4f5a033b2c94b3c7cadcfc96f83).

Your way is a lot faster that "strings.HasPrefix", but using a slice to compare the chars 1:8 is actually faster that comparing each char individually.

With go >=1.7, comparing a slice is always faster, although I could not say if too many optimizations were done when compiling my bench.

Reply:
Thanks, I will take a look. Always enjoy increasing performance.

Reply:
Thanks!